tests/handlers/test_directory.py

```diff
@@ -432,6 +432,48 @@ def test_allowed(self):
         self.assertEquals(200, channel.code, channel.result)
 
 
+class TestCreatePublishedRoomACL(unittest.HomeserverTestCase):
+    user_id = "@test:test"
+    data_template = (
+        '{"room_alias_name": "%%23unofficial_test%%3Atest", "visibility": "%s"}'
+    )
```

Comment:
Does this need to be URL-encoded? I think this is the body of the http request and so should be fine to use `#` and `:` directly.

Having said that, the client-server spec [says](https://spec.matrix.org/unstable/client-server-api/#post_matrixclientr0createroom) that `room_alias_name` is the local part. So I think we just want `unofficial_test` here rather than `#unofficial-test:test`.

```suggestion
    data_template = '{"room_alias_name": "unofficial_test", "visibility": "%s"}'
```

Reply:
Yes, you're right, this is supposed to be just the localpart. That explains why the resolved alias was eventually getting an extra leading hash...!